native/src/main/java/io/ballerina/persist/cmd/Generate.java

```diff
@@ -0,0 +1,252 @@
+            clientPath = Paths.get(this.sourcePath, "modules", "generated_clients",
+                    entity.getEntityName().toLowerCase() + "_client.bal").toAbsolutePath().toString();
+        } else {
+            clientPath = Paths.get(this.sourcePath, "modules", "generated_clients",
+                            entity.getModule() + "_" + entity.getEntityName().toLowerCase() + "_client.bal")
+                    .toAbsolutePath().toString();
+        }
+        try {
+            writeOutputFile(balTree, clientPath);
+        } catch (IOException e) {
+            throw new GenerateBalException("Error occurred in formatting ballerina client object!");
+        } catch (FormatterException e) {
+            throw new GenerateBalException("Error occurred while generating ballerina client object!");
+        }
+
+    }
+    private void generateConfigurationBalFile() throws GenerateBalException {
+        SyntaxTree configTree = BalSyntaxTreeGenerator.generateConfigBalFile();
+        try {
+            writeOutputFile(configTree, Paths.get(this.sourcePath, "modules",
+                            "generated_clients", "database_configuration.bal")
+                    .toAbsolutePath().toString());
+        } catch (IOException e) {
+            throw new GenerateBalException("Error occurred in formatting database configuration file!");
+        } catch (FormatterException e) {
+            throw new GenerateBalException("Error occurred while generating database configuration file!");
+        }
+    }
+
+    private static void writeOutputFile(SyntaxTree syntaxTree, String outPath) throws IOException, FormatterException {
+        String content;
+        content = Formatter.format(syntaxTree.toSourceCode());
+        Path pathToFile = Paths.get(outPath);
+        Files.createDirectories(pathToFile.getParent());
+        try (PrintWriter writer = new PrintWriter(outPath, StandardCharsets.UTF_8.name())) {
+            writer.println(content);
+        }
+    }
+    public static boolean hasSyntacticDiagnostics(Path filePath) {
+        String content;
+        try {
+            content = Files.readString(filePath);
+        } catch (IOException e) {
+            return false;
+        }
+        TextDocument textDocument = TextDocuments.from(content);
+        return SyntaxTree.from(textDocument).hasDiagnostics();
+    }
+
+    public static boolean hasSemanticDiagnostics(Path projectPath,
+                                                 ProjectEnvironmentBuilder projectEnvironmentBuilder) {
+        Package currentPackage;
+        if (projectEnvironmentBuilder == null) {
+            BuildProject buildProject = BuildProject.load(projectPath.toAbsolutePath());
+            currentPackage = buildProject.currentPackage();
+        } else {
+            BuildProject buildProject = BuildProject.load(projectEnvironmentBuilder, projectPath);
+            currentPackage = buildProject.currentPackage();
+        }
```

Comment:
```suggestion
        BuildProject buildProject;
        if (projectEnvironmentBuilder == null) {
            buildProject = BuildProject.load(projectPath.toAbsolutePath());
        } else {
            buildProject = BuildProject.load(projectEnvironmentBuilder, projectPath);
        }
        currentPackage = buildProject.currentPackage();
```

Reply:
Updated in [6d8b4a4](https://github.com/ballerina-platform/persist-tools/pull/15/commits/6d8b4a41b9a0ea1002bb906bb2008380cfa7cdb1)